src/gluonts/evaluation/new_metrics.py

```diff
@@ -0,0 +1,230 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from typing import List
+
+import numpy as np
+
+
+class Stat:
+    name: str
+    dependencies = ()
+
+    def apply(self, x):
+        raise NotImplementedError
+
+
+class Metric:
+    name: str
+    dependencies = ()
+
+    def apply(self, x):
+        x[self.name] = self.loss(x)
+
+    def get_aggr(self, metrics, aggrs, axis):
+        return self.aggregate(metrics[self.name], axis=axis)
+
+    def loss(self, x):
+        raise NotImplementedError
+
+    def aggregate(self, metrics, axis=None):
+        raise NotImplementedError
+
+
+class DerivedMetric:
+    name: str
+    dependencies = ()
+
+    def get_aggr(self, metrics, aggrs, axis):
+        return self.get(aggrs)
+
+    def get(self, aggrs):
+        raise NotImplementedError
+
+
+class Loss(Stat):
+    name: str = "loss"
+
+    def apply(self, x):
+        x.loss = x.target - x.forecast
+
+
+class AggrMean:
+    def aggregate(self, loss, axis=None):
+        return np.mean(loss, axis=axis)
+
+
+class AggrSum:
+    def aggregate(self, loss, axis=None):
+        return np.sum(loss, axis=axis)
+
+
+class AbsTargetSum(AggrSum, Metric):
+    name: str = "abs_target_sum"
+
+    def loss(self, x):
+        return x.target
+
+
+@dataclass
+class QuantileLoss(AggrSum, Metric):
+    q: float
+
+    dependencies = [Loss()]
+
+    @property
+    def name(self):
+        return f"quantile[{self.q}]"
+
+    def loss(self, x):
+        return 2 * np.abs(x.loss * ((x.target <= x.forecast) - self.q))
+
+
+@dataclass
+class WeightedQuantileLoss(DerivedMetric):
+    quantile: float
+
+    @property
+    def name(self):
+        return f"wQuantileLoss[{self.quantile}]"
+
+    @property
+    def dependencies(self):
+        return [QuantileLoss(self.quantile), AbsTargetSum()]
+
+    def get(self, metrics):
+        ql = QuantileLoss(self.quantile)
+
+        return metrics[ql.name] / metrics.abs_target_sum
+
+
+@dataclass
+class MeanWeightedQuantileLoss(DerivedMetric):
+    quantiles: List[float] = (0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9)
+    name: str = "mean_wQuantileLoss"
+
+    @property
+    def dependencies(self):
+        return list(map(WeightedQuantileLoss, self.quantiles))
+
+    def get(self, metrics):
+        return np.mean(
+            [metrics[quantile.name] for quantile in self.dependencies],
+            axis=0,
+        )
+
+
+def merge(a, b):
```

Comment:
It seems that the functions `merge`, `resolve`, and `ger_order` are necessary to keep metric dependencies in a consistent way. Can we somehow group this functions so that they are clearly standing out on their purpose? Otherwise they might look a bit like new metrics.